### ISSUE TITLE
Set current page to paginator in EloquentDataProvider

### ISSUE THREAD

**Issue by Nayjest** [Nayjest/Grids#135]
1) Remove following temporary fix from CsvExport & ExcelExport:

``` php
 $provider->getPaginationFactory()->setPageName('page_unused');
```

2) Check that $dataProvider->setCurrentPage() don't works for EloquentDataProvider  (navigate to second page & click CSV export, get empty file instead of data)

3) (SUGGESTION) Use $dataProvider->setCurrentPage()  inside EloquentDataProvider::getPaginator() to solve issue. (ANOTHER SOLUTION?)

Solution should not affect global pagination state, should not broke another pagination that can exist on page.

**Comment by wdog** [comment]
hi can you explain what's happening. 

I've notice the problem. CsvExport does not work 
removing from the Class CsvExport

```php
$provider->getPaginationFactory()->setPageName('page_unused');
```

works but only on first page.
but then i'm lost...

**Comment by wdog** [comment]
I solved this way, hope it is right

```php

  protected function resetPagination(DataProvider $provider)
  {
    $currentPage = 1;
    Paginator::currentPageResolver(function () use ($currentPage) {
      return $currentPage;
    });
    $provider->setPageSize($this->getRowsLimit());
  }
```

**Comment by Nayjest** [comment]
Hi!
Paginator adds limit & offset to query.
For CSV / Excel export we need to abandon pagination for exporting all data instead of current page.

Implementation that existed in grids package from ancient times:
```php
protected function resetPagination(DataProvider $provider)
      {
          $provider->setPageSize($this->getRowsLimit());
          $provider->setCurrentPage(1);
      }
```

Recently @hexusdev  discovered that resetting pagination doesn't work if there is GET parameter related to pagination in URL. I don't know was it broken with some Laravel updates or bug existed all time. So, he added temporary fix: he renamed GET parameter related to pagination to avoid applying pagination.

Your solution looks better, one more suggestion:
```php
protected function resetPagination(DataProvider $provider)
  {
    Paginator::currentPageResolver(function () {
        return 1;
    });
    $provider->setCurrentPage(1);
    $provider->setPageSize($this->getRowsLimit());
  }
```

But what is bad: both solutions (yours and solution from @hexusdev) affects global pagination state, both solutions will broke another pagination that can exist on page with grid (for example, if we have multiple grids on same page or grid in sidebar that appears on some controller action with pagination)

**Comment by wdog** [comment]
I think that my solution in L5 will not break anything because in https://github.com/Nayjest/Grids/blob/master/src/Components/Laravel5/Pager.php

```php
 Paginator::currentPageResolver(function () {
            return Input::get("$this->input_key.page", 1);
        });
```
Is always redefined when rendered

Am I wrong?

**Comment by Nayjest** [comment]
Yep, looks like you are right.
Bad news: Nayjest/Grids/blob/master/src/Components/Laravel5/Pager isn't good too.
If this will not break pagination for another grids, this may break it for 3-rd party code, becouse it's still uses a global state (Paginator::currentPageResolver).

Example:
1) Developer configures pagination for controller
2) Grid is rednered somewhere, it resets Paginator::currentPageResolver
3) Controller renders page renders with it's own pagination... and whoops

I don't seen that bugs personally, it's only my suggestions.

Laravel pagination makes me mad :)

Therefore I not used it in view-components:

https://github.com/view-components/view-components/blob/master/src/Component/Control/View/PaginationControlView.php
https://github.com/view-components/view-components/blob/master/resources/views/controls/pagination.php
https://github.com/view-components/view-components/blob/master/src/Component/Control/PaginationControl.php

**Comment by wdog** [comment]
hi @Nayjest,  the "collateral effects" on 3rd party code are almost known and can be avoided 
(grids are safe! :+1:)

I hope to find permanent solution, but it will be very hard.

In the next project I'll try to use the new **view-components** but at the moment this can be considered an _half close_ issue.

in any case `getPaginationFactory` in

```php
$provider->getPaginationFactory()->setPageName('page_unused');
```
with L5.1 returns me an `FatalThrowableError`

```
Call to undefined method Illuminate\Database\MySqlConnection::getPaginator()
```

https://laravel.com/api/5.1/Illuminate/Database/MySqlConnection.html

it works only with L4.2
https://laravel.com/api/4.2/Illuminate/Database/MySqlConnection.html

**Comment by Nayjest** [comment]
Oh, did not know about error in L5, I will add version check now & your solution for L5+

**Comment by Nayjest** [comment]
It's becouse no tests in this package :(
In view-components testing helps greatly.

**Comment by Nayjest** [comment]
@wdog can you check [fix135 branch](https://github.com/Nayjest/Grids/tree/fix135) (see [changes](https://github.com/Nayjest/Grids/pull/138/files)) for your application?

If it works fine, I will release it immediately after your response.

**Comment by wdog** [comment]
L5+ is OK!

**Comment by Nayjest** [comment]
Fix for L5 released in [v1.3.1](https://github.com/Nayjest/Grids/releases/tag/v1.3.1)

I don't  close this issue because problems with global pagination state still possible.

**Comment by Nayjest** [comment]
@wdog thanks for assistance!

**Comment by wdog** [comment]
@Nayjest I'm so sorry but in the Csv And Excel Classes  I've not noticed the header of the file

right 

```php
use Illuminate\Pagination\Paginator;
```
not 

```php 
use Paginator;
```

I'm really sorry but I've tasted the method and I've resolved manually the import of the class.

**Comment by wdog** [comment]
I've investigated a little bit, but seems that `Illuminate\Support\Facade\Paginator` has been removed since 4.2

https://github.com/laravel/framework/blob/4.2/src/Illuminate/Support/Facades/Paginator.php

**Comment by Nayjest** [comment]
Ok, if you work with L5+, check master branch please and give me know that it works fine, then I will release it in v1.3.3

**Comment by wdog** [comment]
I've tested in  Laravel  5.1.40, and It works!

**Comment by Nayjest** [comment]
Thanks! Released v1.3.3
